Shared/Samples/Filter features in scene/FilterFeaturesInSceneView.swift

```diff
@@ -0,0 +1,240 @@
+// you may not use this file except in compliance with the License.
+// You may obtain a copy of the License at
+//
+//   https://www.apache.org/licenses/LICENSE-2.0
+//
+// Unless required by applicable law or agreed to in writing, software
+// distributed under the License is distributed on an "AS IS" BASIS,
+// WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+// See the License for the specific language governing permissions and
+// limitations under the License.
+
+import ArcGIS
+import SwiftUI
+
+struct FilterFeaturesInSceneView: View {
+    /// The view model for this sample.
+    @StateObject private var model = Model()
+    
+    /// A Boolean value indicating whether to show an error alert.
+    @State private var isShowingAlert = false
+    
+    /// The error shown in the error alert.
+    @State private var error: Error? {
+        didSet { isShowingAlert = error != nil }
+    }
+    
+    var body: some View {
+        SceneView(scene: model.scene, graphicsOverlays: [model.graphicsOverlay])
+            .task {
+                do {
+                    try await model.scene.load()
+                } catch {
+                    self.error = error
+                }
+            }
+            .toolbar {
+                ToolbarItemGroup(placement: .bottomBar) {
+                    Button(model.filterState.label) {
+                        model.handleFilterState()
+                    }
+                }
+            }
+            .alert(isPresented: $isShowingAlert, presentingError: error)
+    }
+}
+
+private extension FilterFeaturesInSceneView {
+    /// The model used to store the geo model and other expensive objects
+    /// used in this view.
+    class Model: ObservableObject {
+        /// The scene for this sample.
+        let scene: ArcGIS.Scene
+        
+        /// The open street map layer for the sample.
+        private let osmBuildings = ArcGISSceneLayer(
+            item: PortalItem(
+                portal: .arcGISOnline(connection: .anonymous),
+                id: .osmBuildings
+            )
+        )
+        
+        /// The San Francisco buildings scene layer for the sample.
+        private let buildingsSceneLayer = ArcGISSceneLayer(
+            url: URL(string: "https://tiles.arcgis.com/tiles/z2tnIkrLQ2BRzr6P/arcgis/rest/services/SanFrancisco_Bldgs/SceneServer")!
+        )
+        
+        /// The graphics overlay for the scene view.
+        let graphicsOverlay = GraphicsOverlay()
+        
+        /// A polygon that shows the extent of the detailed buildings scene layer.
+        private let polygon: ArcGIS.Polygon = makeFilteringPolygon()
+        
+        /// A red extent boundary graphic that represents the full extent of the detailed buildings scene layer.
+        private let sanFranciscoExtentGraphic: Graphic
+        
+        /// The filter state for the scene view.
+        @Published private(set) var filterState: FilterState = .addBuildings
```

Comment:
Sorry I missed this. As this variable have effects on the UI, the view model [needs](https://developer.apple.com/videos/play/wwdc2021/10019/?time=1300) `@MainActor`. Can you make a quick fix? Thanks.

Reply:
I would consider making the `FilterFeaturesInSceneView.Model` MainActor. I think ObservableObjects usually should be.